marshmallow/marshalling.py

```diff
@@ -174,7 +174,7 @@ def _run_validator(self, validator_func, output,
             # Store or reraise errors
             if err.field_names:
                 field_names = err.field_names
-                field_objs = [fields_dict[each] for each in field_names]
+                field_objs = [fields_dict[each] if each in fields_dict else None for each in field_names]
```

Comment:
We shouldn't put `None` in this list if a field name isn't in `fields_dict`. You might need to do an old-fashioned for-loop to fix this.